internal/pkg/aws/cloudwatchlogs/cloudwatchlogs.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package cloudwatchlogs contains utility functions for Cloudwatch Logs client.
+package cloudwatchlogs
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudwatchlogs"
+)
+
+const (
+	// SleepDuration is the sleep time for making the next request for log events.
+	SleepDuration        = 1 * time.Second
+	orderByLastEventTime = "LastEventTime"
+)
+
+var (
+	fatalCodes   = []string{"FATA", "FATAL", "fatal", "ERR", "ERROR", "error"}
+	warningCodes = []string{"WARN", "warn", "WARNING", "warning"}
+)
+
+type cloudwatchlogsClient interface {
+	DescribeLogStreams(input *cloudwatchlogs.DescribeLogStreamsInput) (*cloudwatchlogs.DescribeLogStreamsOutput, error)
+	GetLogEvents(input *cloudwatchlogs.GetLogEventsInput) (*cloudwatchlogs.GetLogEventsOutput, error)
+}
+
+// Service wraps an AWS Cloudwatch Logs client.
+type Service struct {
+	cwlogs cloudwatchlogsClient
+}
+
+// GetLogEventsOpts sets up optional parameters for LogEvents function.
+type GetLogEventsOpts func(*cloudwatchlogs.GetLogEventsInput)
+
+// WithLimit sets up limit for GetLogEventsInput
+func WithLimit(limit int) GetLogEventsOpts {
+	return func(in *cloudwatchlogs.GetLogEventsInput) {
+		in.Limit = aws.Int64(int64(limit))
+	}
+}
+
+// WithStartTime sets up startTime for GetLogEventsInput
+func WithStartTime(startTime *int64) GetLogEventsOpts {
```

Comment:
nit: these functions should be like `WithLimit` and accept the concrete type `int64` instead of the pointer. That makes it easier to use for the client.